pkg/api/v1beta1/conversion.go

```diff
@@ -116,6 +116,26 @@ func init() {
 			out.Value = in.Value
 			out.Key = in.Name
 			out.Name = in.Name
+
+			if err := s.Convert(&in.EnvVarSource, &out.EnvVarSource, 0); err != nil {
+				return err
+			}
+
+			return nil
+		},
+		func(in *newer.EnvVarSource, out *EnvVarSource, s conversion.Scope) error {
```

Comment:
@thockin This particular one exists because of the v1beta1 `Key` field of env var.